compiler/modules/front/expr.bal

```diff
@@ -1476,26 +1476,17 @@ function registerName(bir:Register register) returns string {
     return <string>register.name;
 }
 
-function codeGenCall(ExprContext cx, bir:BasicBlock curBlock, bir:FunctionRef func, t:SemType returnType, bir:Operand[] args, Position pos) returns ExprEffect {
+function codeGenCall(ExprContext cx, bir:BasicBlock curBlock, bir:FunctionOperand func, 
+                     t:SemType returnType, bir:Operand[] args, boolean restParamIsList, Position pos) returns ExprEffect {
     bir:TmpRegister reg = cx.createTmpRegister(returnType, pos);
-    bir:CallInsn call = {
-        func,
-        result: reg,
-        args: args.cloneReadOnly(),
-        pos
-    };
-    curBlock.insns.push(call);
-    return { result: constifyRegister(reg), block: curBlock };    
-}
-
-function codeGenCallIndirect(ExprContext cx, bir:BasicBlock curBlock, bir:Register func, bir:FunctionRef funcRef, t:SemType returnType, bir:Operand[] args, Position pos) returns ExprEffect {
-    bir:TmpRegister reg = cx.createTmpRegister(returnType, pos);
-    [bir:Register, bir:Operand...] operands = [func, ...args];
-    bir:CallIndirectInsn call = {
-        result: reg,
-        operands: operands.cloneReadOnly(),
-        pos
-    };
+    bir:CallIndirectInsn|bir:CallConstInsn call = func is bir:FunctionConstOperand ?
+                                                    { result: reg,
+                                                      operands: [func, ...args],
+                                                      pos } :
+                                                    { result: reg,
+                                                      operands: [func, ...args],
+                                                      restParamIsList,
+                                                      pos };
```

Comment:
Minor: would be shorter to write if-else.

Reply:
Fixed